### PR TITLE
Have macc_globals object on heap instead of global memory

### PR DESCRIPTION
**Pull request by adizaimi**
Have macc_globals object on heap instead of global memory. 
This structure groups variables used for csc2 parsing and is about 10MB in size--there is no need for it to be in global memory area.

### CODE REVIEW

**Comment by adizaimi** [comment]
cdb2test   Nov 18 22:06:52 2019  in-comdb2db AZ_macc_globals_on_heap.R20191118.6

File: db/comdb2.c [hunk 1]
```diff
@@ -3275,15 +3278,18 @@ static int init_sqlite_table(struct dbenv *dbenv, char *table)
        return -1;
     }
 
+    dyns_init_globals();
     rc = dyns_load_schema_string((char*) schema, dbenv->envname, table);
     if (rc) {
         logmsg(LOGMSG_ERROR, "Can't parse schema for %s\n", table);
-        return -1;
+        rc = -1;
+        goto err;
```
Comment (akshatsikarwar) [inline db/comdb2.c]: I prefer if caller invoked `dyns_init_globals` and invokes `dyns_cleanup_globals` on non-zero return.

File: db/config.c [hunk 2]
```diff
@@ -966,16 +966,19 @@ static int read_lrl_option(struct dbenv *dbenv, char *line,
                     return -1;
                 }
             }
+            dyns_init_globals();
             rc = dyns_load_schema(fname, (char *)gbl_dbname, tblname);
             if (rc != 0) {
                 logmsg(LOGMSG_ERROR, "Error loading %s schema.\n", tok);
+                dyns_cleanup_globals();
```
Comment (akshatsikarwar) [inline db/config.c]: Same - I prefer if we don't sprinkle error handling everywhere.

Comment (riverszhang89) [reply db/config.c]: Agreed.

File: db/comdb2.c (hunk 1 above)
Comment (adizaimi) [reply db/comdb2.c]: Well, in this case i tried to keep them as close as possible together that way it appeared as a pattern: dyns_init_globals, dyns_load_schema, dyns_cleanup_globals. 
Anyway, I did reorganize them as much as i could so now most callers will init and cleanup.
There are however places where we do get a lock and this globals init and cleanup needs to be done while the lock is held so I can't clean them all like you wished.

**Pull request merged**